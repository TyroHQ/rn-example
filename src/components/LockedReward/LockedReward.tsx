import React from "react";
import { Text, View } from "react-native";
import styled from "styled-components";
import { Placeholder } from "../Placeholder";
import { RewardTileHeader } from "../RewardTileHeader";
import { SmallPointStatusBox } from "../SmallPointStatusBox";

const LockedRewardContainer = styled(View)`
  background-color: lightgrey;
  padding: 10px;
`;

export class LockedReward extends React.Component<{}> {
  static defaultProps: LockedReward["props"];
  render() {
    return (
      <Placeholder>
        LockedReward container
        <RewardTileHeader />
        <Placeholder>
          White container
          <SmallPointStatusBox />
          <SmallPointStatusBox />
          <SmallPointStatusBox />
        </Placeholder>
      </Placeholder>
    );
  }
}
