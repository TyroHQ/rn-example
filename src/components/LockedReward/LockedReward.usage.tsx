import React from "react";
import { boolean, text, number } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";
import { LockedReward } from "./";

export const basic = () => <LockedReward />;
