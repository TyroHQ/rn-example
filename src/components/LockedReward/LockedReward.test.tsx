import { shallow } from "enzyme";
import * as usage from "./LockedReward.usage";

describe("<LockedReward />", () => {
  for (const example in usage) {
    it(`renders well in ${example} mode`, () => {
      const x = (usage as any)[example]();
      const wrapper = shallow(x);
      expect(wrapper).toMatchSnapshot();
    });
  }
});
