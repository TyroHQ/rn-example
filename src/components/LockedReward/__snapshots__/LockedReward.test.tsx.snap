// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LockedReward /> renders well in basic mode 1`] = `
<Placeholder
  orientation="row"
>
  LockedReward container
  <RewardTileHeader />
  <Placeholder
    orientation="row"
  >
    White container
    <SmallPointStatusBox />
    <SmallPointStatusBox />
    <SmallPointStatusBox />
  </Placeholder>
</Placeholder>
`;
